joomla/joomla-cms
# [4] When item doesn't exist, show specific message

Closes #33410

### Summary of Changes

This PR makes two changes. 

1) Correctly outputs a specific error message if an item cannot be loaded from the db - whereas presently that message is missing (only a trailing period displays currently) 

2) If the above message is being displayed, then dont also confuse a user and tell them they dont have permission to that use the link. Its not correct. That message is raised in the display controller way after the first message in the models.  There was a previous error before that, in the model/db that is more important (one example is the message above, saying the row doesn't exist, but other messages could exceptions "Table has multiple primary keys specified", "No table keys defined", "Missing field in database", or the `$result` could have been modified by plugins using the `onTableAfterLoad` trigger - and so telling the user after one of these more serious messages, that they "are not permitted to use that link directly" is actually not quite correct. The first message is more important. 

### Testing Instructions

Do some things that create audit log in User Action Logs
Lets say, create a User Note with title "myNoteTitle", save it, trash it, remove it from the trash. 
Then go to Home Dashboard or Users -> User Action Log
see:
`
User admin added new user note myNoteTitle
`
click on "myNoteTitle" hyperlink

Other Example URLS of things you can try 

http://127.0.0.1:4444/administrator/index.php?option=com_content&task=article.edit&id=9999999999999
http://127.0.0.1:4444/administrator/index.php?option=com_users&task=user.edit&id=9999999999999
http://127.0.0.1:4444/administrator/index.php?option=com_users&task=note.edit&id=9999999999999
http://127.0.0.1:4444/administrator/index.php?option=com_banners&task=banner.edit&id=9999999999
http://127.0.0.1:4444/administrator/index.php?option=com_fields&task=field.edit&id=9999999999999&context=com_users.user

### Actual result BEFORE applying this Pull Request

A message telling me I dont have permission to use that link. But note the missing REASON on the first line, before the trailing period.

<img width="636" alt="Screenshot 2021-04-29 at 18 09 40" src="https://user-images.githubusercontent.com/400092/116591339-99d24100-a916-11eb-94db-78eeca363b4d.png">


### Expected result AFTER applying this Pull Request

<img width="1606" alt="Screenshot 2021-04-29 at 20 43 25" src="https://user-images.githubusercontent.com/400092/116608849-92696280-a92b-11eb-8fc8-03b5735544a0.png">


### Documentation Changes Required
